src/ui/connectiondialog.cc

```diff
@@ -0,0 +1,174 @@
+  server_file_dialog_ = new QFileDialog(this);
+  server_file_dialog_->setAcceptMode(QFileDialog::AcceptOpen);
+  server_file_dialog_->setFileMode(QFileDialog::ExistingFile);
+
+  connect(ui_->select_server_executable_button, &QPushButton::clicked,
+      server_file_dialog_, &QFileDialog::show);
+  connect(server_file_dialog_, &QFileDialog::fileSelected,
+      this, &ConnectionDialog::serverFileSelected);
+
+  serverLocalhost();
+  clientLocalhost();
+  randomKey();
+  userAsClientName();
+
+  ui_->database_line_edit->setText("hack-o-store.dark-matter");
+  newServerToggled(ui_->new_server_radio_button->isChecked());
+}
+
+ConnectionDialog::~ConnectionDialog() {
+  db_file_dialog_->deleteLater();
+  delete ui_;
+}
+
+bool ConnectionDialog::runANewServer() {
+  return ui_->new_server_radio_button->isChecked();
+}
+
+QString ConnectionDialog::serverHost() {
+  return ui_->server_host_line_edit->text();
+}
+
+int ConnectionDialog::serverPort() {
+  return ui_->port_spin_box->value();
+}
+
+QString ConnectionDialog::clientInterface() {
+  return ui_->client_interface_line_edit->text();
+}
+
+QString ConnectionDialog::authenticationKey() {
+  return ui_->authentication_key_label->text();
+}
+
+QString ConnectionDialog::clientName() {
+  return ui_->client_name_line_edit->text();
+}
+
+QString ConnectionDialog::databaseFile() {
+  return ui_->database_line_edit->text();
+}
+
+QString ConnectionDialog::serverScript() {
+  return ui_->server_executable_line_edit->text();
+}
+
+void ConnectionDialog::serverLocalhost() {
+  ui_->server_host_line_edit->setText("127.0.0.1");
+}
+
+void ConnectionDialog::clientLocalhost() {
+  ui_->client_interface_line_edit->setText("127.0.0.1");
+}
+
+void ConnectionDialog::randomKey() {
+  // TODO: This is cryptographically-secure on all modern OS-es, but this isn't
+  // explicitely guarateed by the standard. We should fix it someday.
+  std::random_device rd;
+  std::uniform_int_distribution<uint32_t> uniform;
+  auto gen_key_part = [&rd, &uniform](){
+    return QString("%1").arg(uniform(rd), 8 /* width */, 16 /* base */,
+                             QChar('0'));
+  };
+  ui_->key_line_edit->setText(gen_key_part() + gen_key_part() + gen_key_part()
+                              + gen_key_part());
+}
+
+void ConnectionDialog::userAsClientName() {
+  QString client_name;
+#if defined(Q_OS_LINUX) || defined(Q_OS_MAC)
+  client_name = qgetenv("USER");
+#else // Windows
+  client_name = qgetenv("USERNAME");
+#endif
+
+  if(client_name.length() == 0) {
+    client_name = "Veles UI";
+  }
+
+  ui_->client_name_line_edit->setText(client_name);
+}
+
+void ConnectionDialog::newServerToggled(bool toggled) {
+  ui_->database_label->setEnabled(toggled);
+  ui_->database_line_edit->setEnabled(toggled);
+  ui_->select_database_button->setEnabled(toggled);
+
+  ui_->server_executable_label->setEnabled(toggled);
+  ui_->server_executable_line_edit->setEnabled(toggled);
+  ui_->select_server_executable_button->setEnabled(toggled);
+}
```

Comment:
We should also disable "random" button for key generation when you're connecting to an existing server (it doesn't make any sense then).